lib/devicetrust/native/device_windows.go

```diff
@@ -480,6 +476,48 @@ func solveTPMEnrollChallenge(
 	}, nil
 }
 
+func solveTPMAuthDeviceChallenge(
+	challenge *devicepb.TPMAuthenticateDeviceChallenge,
+) (*devicepb.TPMAuthenticateDeviceChallengeResponse, error) {
+	akPath, err := setupDeviceStateDir(os.UserHomeDir)
+	if err != nil {
+		return nil, trace.Wrap(err, "setting up device state directory")
+	}
+
+	tpm, err := attest.OpenTPM(&attest.OpenConfig{
+		TPMVersion: attest.TPMVersion20,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err, "opening tpm")
+	}
+	defer tpm.Close()
```

Comment:
Nit: perhaps log the `err` that `Close()` can return. Given that TMP can be backed by hardware of unknown quality, some useful logs may only be found at this point.

Reply:
3ebf26324f